src/Azure.Functions.Cli/Actions/DeployActions/Platforms/KnativePlatform.cs

```diff
@@ -0,0 +1,147 @@
+using System;
+using System.Linq;
+using System.IO;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Azure.Functions.Cli.Common;
+using Azure.Functions.Cli.Helpers;
+using Azure.Functions.Cli.Interfaces;
+using Colors.Net;
+using KubeClient;
+using KubeClient.Models;
+using Azure.Functions.Cli.Actions.DeployActions.Platforms.Models;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json;
+
+namespace Azure.Functions.Cli.Actions.DeployActions.Platforms
+{
+    public class KnativePlatform : IHostingPlatform
+    {
+        private string configFile = string.Empty;
+        private const string FUNCTIONS_NAMESPACE = "azure-functions";
+        private static KubeApiClient client;
+
+        public async Task DeployContainerizedFunction(string functionName, string image, int min, int max)
+        {
+            await Deploy(functionName, image, FUNCTIONS_NAMESPACE, min, max);
+        }
+
+        public KnativePlatform(string configFile)
+        {
+            this.configFile = configFile;
+            KubeClientOptions options;
+
+            if (!string.IsNullOrEmpty(configFile))
+            {
+                options = K8sConfig.Load(configFile).ToKubeClientOptions(defaultKubeNamespace: FUNCTIONS_NAMESPACE);
+            }
+            else
+            {
+                options = K8sConfig.Load().ToKubeClientOptions(defaultKubeNamespace: FUNCTIONS_NAMESPACE);
+            }
+
+            client = KubeApiClient.Create(options);
+        }
+
+        private async Task Deploy(string name, string image, string nameSpace, int min, int max)
+        {
+            var isHTTP = IsHTTPTrigger(name);
+
+            await CreateNamespace(nameSpace);
+            client.DefaultNamespace = nameSpace;
+
+            ColoredConsole.WriteLine();
+            ColoredConsole.WriteLine("Deploying function to Knative...");
+
+            var knativeService = GetKnativeService(name, image, nameSpace, min, max, isHTTP);
+            var json = Newtonsoft.Json.JsonConvert.SerializeObject(knativeService,
+                            Newtonsoft.Json.Formatting.None,
+                            new Newtonsoft.Json.JsonSerializerSettings
+                            {
+                                NullValueHandling = Newtonsoft.Json.NullValueHandling.Ignore
+                            });
+
+            File.WriteAllText("deployment.json", json);
+            await KubernetesHelper.RunKubectl($"apply -f deployment.json");
+            File.Delete("deployment.json");
+
+            var externalIP = await GetIstioClusterIngressIP();
+            if (externalIP == "")
+            {
+                ColoredConsole.WriteLine("Couldn't find Istio Cluster Ingress External IP");
+                return;
+            }
+
+            var host = GetFunctionHost(name, nameSpace);
+
+            ColoredConsole.WriteLine("");
```

Comment:
nit: you don't need the `""`

```suggestion
            ColoredConsole.WriteLine();
```